src/components/DatePickerInput/DatePickerInput.tsx

```diff
@@ -52,17 +63,29 @@ function DatePickerInput(
                             onPress={onPressCalendarIcon}
                             disabled={disabled}
                         />
-
-                        <DatePickerInputModal
-                            date={value}
-                            mode="single"
-                            isOpen={visible}
-                            onClose={onDismiss}
-                            onConfirm={onInnerConfirm}
-                            locale={locale}
-                            // dateMode={inputMode}
-                            validRange={validRange}
-                        />
+                        {pickerMode === 'modal' ? (
```

Comment:
rightElement can be a separate function outside the main component. with the necessary props either passed.

The idea is to write a better way this the ternary operator logic we have here.


eg:
```

    // either in the body of of this component or a separate renderer function.
   const renderers = {
       modal: () => <DatePickerInputModal
                                date={value}
                                mode="single"
                                isOpen={visible}
                                onClose={onDismiss}
                                onConfirm={onInnerConfirm}
                                locale={locale}
                                // dateMode={inputMode}
                                validRange={validRange}
                            />,
                            
        docked: () => <DatePickerDocked
                                date={value}
                                locale={locale}
                                startYear={startYear}
                                endYear={endYear}
                                onChange={onInnerConfirm}
                                isOpen={isOpen}
                                onToggle={onToggle}
                                triggerRef={triggerRef}
                            />
   }
   
   
   renderers[type]()
```

Reply:
resolve @shashank1010